applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -123,18 +129,37 @@ LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_CONFIG_CHANNEL_DFU_LOG_LEVEL);
 		#define MCUBOOT_PRIMARY_SLOT_ID   DT_FIXED_PARTITION_ID(MCUBOOT_PRIMARY_NODE)
 		#define MCUBOOT_SECONDARY_SLOT_ID DT_FIXED_PARTITION_ID(MCUBOOT_SECONDARY_NODE)
 
-		/* Use range check to allow for placing MCUboot header in a separate partition,
-		 * so the application code partition is not an alias for the MCUboot partition,
-		 * but a subpartition of the MCUboot partition.
-		 */
-		#if (CODE_PARTITION_START_ADDR >= MCUBOOT_PRIMARY_START_ADDR) && \
-		    (CODE_PARTITION_START_ADDR < MCUBOOT_PRIMARY_END_ADDR)
-			#define DFU_SLOT_ID MCUBOOT_SECONDARY_SLOT_ID
-		#elif (CODE_PARTITION_START_ADDR >= MCUBOOT_SECONDARY_START_ADDR) && \
-		      (CODE_PARTITION_START_ADDR < MCUBOOT_SECONDARY_END_ADDR)
-			#define DFU_SLOT_ID MCUBOOT_PRIMARY_SLOT_ID
+
+		#if CONFIG_MCUBOOT_BOOTLOADER_MODE_RAM_LOAD
+			/* The RAM load mode assumes a specific RAM layout in DTS. */
+			#define APP_RXM_REGION_NODE     DT_NODELABEL(cpuapp_sram_app_rxm_region)
+			#define MCUBOOT_RAM_REGION_NODE DT_NODELABEL(cpuapp_sram_mcuboot_ram_region)
+
+			BUILD_ASSERT(DT_NODE_EXISTS(APP_RXM_REGION_NODE),
+				"Missing RXM (ROM + RAM) region definition "
+				"(cpuapp_sram_app_rxm_region) for the application image in DTS.");
+			BUILD_ASSERT(DT_NODE_EXISTS(MCUBOOT_RAM_REGION_NODE),
+				     "Missing RAM region definition "
+				     "(cpuapp_sram_mcuboot_ram_region) for the MCUboot bootloader "
+				     "in DTS.");
+			BUILD_ASSERT(DT_REG_ADDR(MCUBOOT_RAM_REGION_NODE) >=
```

Comment:
Consider avoiding making these checks specific to config channel DFU image transport (the checks are relevant also for other DFU image transports)